hw/ip/spi_device/data/spi_device.hjson

```diff
@@ -669,6 +677,259 @@
+            If 0, only one Locality is provided
+            '''
+        } // f: locality
+        { bits: "18:16"
+          name: "max_xfer_size"
+          desc: '''The maximum transfer size in bytes the TPM submodule supports.
+            The value is the exponent of the 2.
+
+            - 3'b 010: Support up to 4B
+            - 3'b 011: Support up to 8B
+            - 3'b 101: Support up to 32B
+            - 3'b 110: Support up to 64B
+
+            All other values are reserved.
+
+            The SW may advertises as supporting more than `max_xfer_size` to the South Brige.
+            If the SW reports the transfer size more than the value described in this field,
+            the SW must read the data from the write FIFO before the FIFO becomes full in order not to make FIFO overflow.
+            '''
+        } // f: max_xfer_size
+      ]
+      tags: [
+        // Remove Read check (reset value)
+        "excl:CsrAllTests:CsrExclCheck"]
+    } // R: TPM_CAP
+    { name: "TPM_CFG"
+      desc: '''TPM Configuration register.
+        '''
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "0"
+          name: "en"
+          desc: "If 1, TPM submodule accepts the transactions over SPI"
+        } // f: en
+        { bits: "1"
+          name: "hw_reg_dis"
+          desc: '''If 0, TPM submodule directly returns the return-by-HW registers for the read requests.
+
+            If 1, TPM submodule uploads the TPM command regardless of the address, and the SW may return the value through the read FIFO.
+            '''
+        } // f: hw_reg_dis
+        { bits: "2"
+          name: "invalid_ool"
+          desc: '''If 1, TPM submodule returns the invalid data (0xFF) for the out of the max Locality request.
+            If it is a write request, it discards the request.
+
+            If 0, TPM submodule uploads the TPM command and address. The SW may write 0xFF to the read FIFO.
+
+            Note: The TPM submodule uploads the TPM commands that do not fall into the FIFO registers (0xD4_XXXX) regardless of `invalid_ool` bit.
+            '''
+        } // f: invalid_ool
+      ]
+    } // R: TPM_CFG
+    { name: "TPM_STATUS"
+      desc: '''TPM submodule state register.
+
+        The TPM_STATUS CSR provides the current TPM status, mostly the buffer and FIFO status.
+        '''
+      swaccess: "ro"
+      hwaccess: "hwo"
+      fields: [
+        { bits: "0"
+          name: "cmdaddr_notempty"
+          desc: "If 1, the TPM_CMD_ADDR has a valid data. This status is reported via the interrupt also."
+        } // f: cmdaddr_notempty
+        { bits: "1"
+          name: "rdfifo_notempty"
+          desc: "If 1, the TPM_READ_FIFO still has pending data."
+        } // f: rdfifo_notempty
+        { bits: "6:4"
+          name: "rdfifo_depth"
+          desc: "This field represents the current read FIFO depth"
+        } // f: rdfifo_depth
+        { bits: "10:8"
```

Comment:
do you think we need a status to catch bad host behavior? let's say the host for some reason sends 65B, which I assume we will discard?

Reply:
The logic fills the FIFO up to the xfer_size then discard. The xfer_size supports up to 64B. If the FIFO becomes full, I think it overwrites? (let me check the design again)

Reply:
o okay sounds good.  Maybe good to check with software then to see what they prefer as a debugging tool in case something like this happens.  I honestly don't think it matters too much exactly what we do as long as there is an indication something like this happened.